src/pages/Typing/Typing.tsx

```diff
@@ -0,0 +1,135 @@
+import { Fragment, FC, useState } from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+import { Container, Paper, Grid, TextField } from '@material-ui/core'
+import { TypingHeader, Typography, Result, Loader } from '../../components'
+import { typingStyles } from './Typing.styles'
+import { saveTestResult } from '../../redux'
+
+interface TypingProps {}
+
+const Typing: FC<TypingProps> = () => {
+  const {
+    testData: { testInfo: { paragraph = '', level = '', time = 0 } = {} } = {},
+    userLogin: { userInfo: { displayName = '', photoURL = '' } = {} } = {},
+  } = useSelector((state: any) => state)
+  // const [wordsArray, setWordsArray] = useState<any>([])
+  const dispatch = useDispatch()
+  const classes = typingStyles()
+
+  const [correctEntries, setCorrectEntries] = useState<number>(0)
+  const [inCorrectEntries, setInCorrectEntries] = useState<number>(0)
+  const [greenArray, setGreenArray] = useState<Array<number>>([])
+  const [redArray, setRedArray] = useState<Array<number>>([])
+  const [stop, setStop] = useState<boolean>(false)
+  const [result, setResult] = useState({
+    accuracy: 0,
+    netWPM: 0,
+  })
+
+  // useEffect(() => {
+  //   const wordArr = paragraph.split(' ')
+  //   setWordsArray(wordArr)
+  // }, [paragraph])
+  const handleText = ({ target: { value = '' } = {} }) => {
```

Comment:
Iam not sure, but i think this component will re render a lot of times. So would it be a good idea to use useCallback, for function definitions in this component?